### PR TITLE
[ch45300] Make BulkMaxDocs and BulkMaxBuffer callbacks so we can flag them

### PR DESCRIPTION
**Pull request by samstokes**
We want to be able to tweak these params. This lets us specify a callback so we can evaluate a flag each time.

### CODE REVIEW

File: lib/corebulk.go [hunk 1]
```diff
@@ -113,12 +113,16 @@ func (c *Conn) NewBulkIndexer(maxConns int) *BulkIndexer {
...
+}
+
+func (c *Conn) NewBulkIndexerWithParams(maxConns, bulkChannelSize int, bulkMaxBufferCb, bulkMaxDocsCb func() int) *BulkIndexer {
 	b := BulkIndexer{conn: c, sendBuf: make(chan *bytes.Buffer, maxConns)}
 	b.needsTimeBasedFlush = true
 	b.buf = new(bytes.Buffer)
 	b.maxConns = maxConns
-	b.BulkMaxBuffer = BulkMaxBuffer
-	b.BulkMaxDocs = BulkMaxDocs
+	b.BulkMaxBufferCB = bulkMaxBufferCb
```
Comment (zsiddique) [inline lib/corebulk.go:124]: Should we set a default value?  Just to be safe if say 0 is passed in

File: lib/corebulk.go [hunk 2]
```diff
@@ -79,9 +79,9 @@ type BulkIndexer struct {
 	// Buffer for Max number of time before forcing flush
 	BufferDelayMax time.Duration
 	// Max buffer size in bytes before flushing to elasticsearch
-	BulkMaxBuffer int // 1048576
+	BulkMaxBufferCB func() int // 1048576
```
Comment (zsiddique) [inline lib/corebulk.go]: Maybe remove the default comment

File: lib/corebulk.go (hunk 1 above)
Comment (samstokes) [reply lib/corebulk.go:124]: You mean if someone passes `func() int { return 0 }`? The default if you don't specify a callback at all (i.e. if you keep using the `NewBulkIndexerWithChannelSize` constructor that doesn't take a callback) is to [use the existing default](https://github.com/launchdarkly/elastigo/pull/20/files#diff-aa16db2eed67a9c127da15da91d60aadR116) (of 16384 bytes).

File: lib/corebulk.go (hunk 2 above)
Comment (samstokes) [reply lib/corebulk.go]: Gah, the default comment is also wrong (the default is 16kB, not 1MB as that suggests). Will remove.

File: lib/corebulk.go (hunk 1 above)
Comment (zsiddique) [reply lib/corebulk.go:124]: Oh derp I did not see you had a new function.  M

**Review by zsiddique — APPROVED** [review]

**Comment by jstncbllr** [comment]
lgtm too

**Pull request merged**